src/haddock/modules/analysis/alascan/__init__.py

```diff
@@ -136,26 +135,42 @@ def _run(self):
                     results_by_model[result.model_id].append(result)
 
             # Save to .tsv
-            for model_id, results in results_by_model.items():
-                write_scan_out(results, model_id)
+            scan_writter_jobs = [
+                GenericTask(write_scan_out, results, model_id)
+                for model_id, results in results_by_model.items()
+            ]
+            engine = Engine(scan_writter_jobs)
+            engine.run()
 
             # Cluster-based analysis
-            clt_alascan = alascan_cluster_analysis(models)
-            
-            # Generate plots if requested
-            if self.params["plot"]:
-                create_alascan_plots(
-                    clt_alascan,
-                    self.params["scan_residue"],
+            clt_scan, clt_pops = group_scan_by_cluster(models)
+            alascan_cluster_jobs = [
+                ClusterOutputer(
+                    clt_data,
+                    clt_id,
+                    clt_pops[clt_id],
+                    scan_residue=self.params["scan_residue"],
+                    generate_plot=self.params["plot"],
                     offline=self.params["offline"],
                 )
+                for clt_id, clt_data in clt_scan.items()
+            ]
+            engine = Engine(alascan_cluster_jobs)
+            engine.run()
```

Comment:
quick comment: if I understand the logic, this will first write many files in parallel and second, group these files also in parallel.

if that is the case (i did not look much into the code), then you are using the filesystem to store temporary information - this is unnecessary and add a lot of I/O.

I have already added the logic to the scheduler to retrieve results directly from the tasks, to avoid exactly this type of issue (using the filesystem as a channel)

https://github.com/haddocking/haddock3/blob/69128756ea5d7fab3af10aa9b255651561bf1483/src/haddock/libs/libparallel.py#L175-L190

there is an implementation that uses this design here:

https://github.com/haddocking/haddock3/blob/69128756ea5d7fab3af10aa9b255651561bf1483/src/haddock/modules/analysis/caprieval/__init__.py#L221-L225